File: Hilos Java/Ficheros texto_binario/Binario.java
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

public class Binario {
     public static void main(String[] args) {
        Producto_Binario producto = new Producto_Binario("Laptop", 2000);

        String archivo = "producto.dat";

        //Serialización
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(archivo))){
            oos.writeObject(producto);
            System.out.println("Objeto serailizado y guardado en :" +archivo);
        } catch (Exception e) {
            // TODO: handle exception
        }

        //Deserialización
        try(ObjectInputStream ois = new ObjectInputStream(new FileInputStream(archivo))) {
            Producto_Binario productoLeido = (Producto_Binario) ois.readObject();
            System.out.println("Objeto Deserializado: "+productoLeido);
            
        } catch (Exception e) {
            // TODO: handle exception
        }
     }
}

File: Ficheros texto_binario/Binario.py
import pickle

class Binario:
    def __init__(self, nombre, precio):
        self.nombre = nombre
        self.precio = precio
    
    def __str__(self):
        return f"Producto: {self.nombre}, Precio: {self.precio}"

producto = Binario("Laptop", 2000)
archivo = "producto.dat"

# Serialización
try:
    with open(archivo, "wb") as file:
        pickle.dump(producto, file)
    print(f"Objeto serializado y guardado en: {archivo}")
except Exception as e:
    print(f"Error en la serialización: {e}")

# Deserialización
try:
    with open(archivo, "rb") as file:
        producto_leido = pickle.load(file)
    print(f"Objeto deserializado: {producto_leido}")
except Exception as e:
    print(f"Error en la deserialización: {e}")
